prelude.links

```diff
@@ -1166,3 +1166,7 @@ fun assertEq (actual, expected) server {
       sysexit(2);
   }
 }
+
+####
+sig addRoute : (String, (String) ~> Page) ~> ()
```

Comment:
This "addRoute" / "startServer" stuff seems awfully imperative. What happens if I call startServer twice?  What happens if I call addRoute after startServer?  Given that Links is supposedly functional, is there an obstacle I'm not seeing to defining a "functional" startServer that takes a list of pairs (String, String ~> Page) and starts the server using the list as the route table?  Or is there some benefit to the imperative API that I'm not seeing?